firmware/pico/src/app/hal.hh

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <cstdint>
+
+uint64_t get_time_since_boot_us();
+uint32_t get_time_since_boot_ms();
+
+class ClockSource {
+   public:
+    static ClockSource& instance();
+    ~ClockSource();
+
+    uint64_t get48MHzTickCount();
+    uint64_t get12MHzTickCount();
+
+   private:
+    static void handleIrq();
+
+    static volatile uint32_t _interruptCount;
```

Comment:
Member variables should be [snake case and followed with a trailing underscore](https://google.github.io/styleguide/cppguide.html#Variable_Names), for instance `interrupt_count_`).

I've been declaring member variables after functions (order goes constants, functions, variable names). Not sure if that's the industry standard but it would be nice to maintain consistency.